### PR TITLE
Resolve all Page Meta by using optional resolvers

### LINKED ISSUE #3186: Equip Page Meta resolvers with more standard functionality
Currently, individual page meta resolvers aggregate the metadata by implementing the `resolve()` method. This method is responsible for calling different resolvers with the required data. This makes it cumbersome to extend, since whenever you like to extend the list of resolvers, you must override (and reimplement) the resolve method. Moreover, the data required by different resolvers might differ, which leads also to reimplementing the `resolve` method. 

We like to allow for easy implementation of those meta resolvers. This means that 
- the `PageMetaService` should call the individual resolvers (if available) on the resolver class
- the individual resolvers must gather the data themselves; we no longer will pass in data through arguments

This change will lead to a breaking changes in the API, which we cannot before 2.0. We must either postpone this until 2.0 or have a temporary backwards compatible feature in place. 

Stretch goals:
- make the list of resolvers configurable for the `PageMetaService`, so that custom/future resolvers can be added without touching the implementation cross all resolvers
- execute some resolvers only for SSR or dev mode, to further optimise the runtime

### PR DESCRIPTION
**Pull request by tobi-or-not-tobi**
Allows for easy extensibility of Page Meta resolvers.

closes #3186

### CODE REVIEW

File: projects/core/src/cms/facade/page-meta.service.ts [hunk 1]
```diff
@@ -34,8 +55,81 @@ export class PageMetaService {
...
+        )
+      );
+    }
+  }
+
+  /**
+   * During release 1.x, we cannot simply return all assigned data
+   * since some (custom) data streams still respond string values
+   */
+  private resolveWhileWereIn1x(data: any[]): {} {
```
Comment (dunqan) [inline projects/core/src/cms/facade/page-meta.service.ts]: What about `resolveFor1x`? ;-)

File: projects/core/src/cart/services/cart-page-meta.resolver.ts [hunk 2]
```diff
@@ -15,27 +20,67 @@ import { PageType } from '../../model/cms.model';
...
-  resolve(): Observable<PageMeta> {
-    return this.cms.getCurrentPage().pipe(
-      filter(page => page !== undefined),
+  /**
+   * The resolve method is no longer preferred and will be removed with release 2.0.
+   * The caller `PageMetaService` service is improved to expect all individual resolvers
+   * instead, so that the code is easier extensible.
+   *
+   * @param skip indicates that this method is not used. While this flag is used by the
+   * calling `PageMetaService`, it is not ysed by custom subclasses when they call their `super`.
```
Comment (dunqan) [inline projects/core/src/cart/services/cart-page-meta.resolver.ts]: Typo for `ysed`

File: projects/core/src/cms/page/content-page-meta.resolver.ts [hunk 3]
```diff
@@ -13,6 +13,17 @@ import { PageBreadcrumbResolver, PageTitleResolver } from './page.resolvers';
 })
 export class ContentPageMetaResolver extends PageMetaResolver
   implements PageTitleResolver, PageBreadcrumbResolver {
+  /**
+   * Backwards compatibility is quarenteed with this flag during
```
Comment (dunqan) [inline projects/core/src/cms/page/content-page-meta.resolver.ts]: typo for `quarenteed`

File: projects/core/src/cart/services/cart-page-meta.resolver.ts [hunk 4]
```diff
@@ -2,39 +2,75 @@ import { Injectable } from '@angular/core';
...
+  /**
+   * @deprecated since version 1.3
+   * With 2.0, the argument(s) will be removed and the return type will change.
+   */
+  resolveTitle(page?: Page): Observable<string> {
+    return page ? of(page.title) : this.cms$.pipe(map(p => p.title));
   }
 
+  /**
+   * @deprecated since version 1.3
```
Comment (dunqan) [inline projects/core/src/cart/services/cart-page-meta.resolver.ts]: 'resolveRobots(): Observable<PageRobotsMeta[]>` is not deprecated, yup?

Comment (tobi-or-not-tobi) [reply projects/core/src/cart/services/cart-page-meta.resolver.ts]: Correct, same is true for the checkout resolver. Only the args aren't expected anymore. I've removed the deprecation note.

File: projects/core/src/cms/facade/page-meta.service.spec.ts [hunk 5]
```diff
@@ -48,16 +53,47 @@ class AnotherPageResolver extends PageMetaResolver {
     this.pageTemplate = 'template';
   }
 
-  resolve(): Observable<PageMeta> {
+  resolve(_skip?: boolean): Observable<PageMeta> | any {
```
Comment (dunqan) [inline projects/core/src/cms/facade/page-meta.service.spec.ts]: This change is probably a left over

**Pull request merged**